src/lib/logger.js

```diff
@@ -46,42 +57,79 @@ function setup(logs) {
       return JSON.stringify({...obj, msg}, Logger.safeCycles()) + '\n';
     };
 
-    if (target.type === 'file') {
-      // destination stream
-      dest = require('fs').createWriteStream(target.path, {flags: 'a', encoding: 'utf8'});
-      dest.on('error', function(err) {
-        Logger.emit('error', err);
-      });
-    } else if (target.type === 'stdout' || target.type === 'stderr') {
-      dest = target.type === 'stdout' ? process.stdout : process.stderr;
-      destIsTTY = dest.isTTY;
-    } else {
-      throw Error('wrong target type for a log');
+    let level = target.level || 35;
+    if (level === 'http') {
+      level = 35;
     }
 
-    if (target.format === 'pretty') {
-      // making fake stream for prettypritting
-      stream.write = (obj) => {
-        dest.write(prettyPrint(obj));
-      };
-    } else if (target.format === 'pretty-timestamped') {
-      // making fake stream for prettypritting
-      stream.write = (obj) => {
-        dest.write(prettyTimestampedPrint(obj));
-      };
+    // create a stream for each log configuration
+    if (target.type === 'rotating-file') {
+      if (target.format !== 'json') {
+        throw new Error('Rotating file streams only work with JSON!');
+      }
+
+      const stream = new VerdaccioRotatingFileStream(
+        _.merge({},
+          // Defaults can be found here: https://github.com/trentm/node-bunyan#stream-type-rotating-file
+          target.options || {},
+          {path: target.path, level})
+      );
+
+      streams.push(
+          {
+            type: 'raw',
+            level,
+            stream,
+          }
+        );
     } else {
-      stream.write = (obj) => {
-        dest.write(jsonPrint(obj));
+      const stream = new Stream();
+      stream.writable = true;
+
+      let dest;
+      let destIsTTY = false;
+      const prettyPrint = (obj) => print(obj.level, obj.msg, obj, destIsTTY) + '\n';
+      const prettyTimestampedPrint = (obj) => obj.time.toISOString() + print(obj.level, obj.msg, obj, destIsTTY) + '\n';
+      const jsonPrint = (obj) => {
+        const msg = fillInMsgTemplate(obj.msg, obj, destIsTTY);
+        return JSON.stringify({...obj, msg}, Logger.safeCycles()) + '\n';
       };
-    }
 
+      if (target.type === 'file') {
```

Comment:
define `dest` here. I'll suggest using full name instead of short forms. `destination`